src/overlays/actors/ovl_Demo_6K/z_demo_6k.c

```diff
@@ -27,57 +54,781 @@ const ActorInit Demo_6K_InitVars = {
+        this->unk_168 = D_8096930C[this->timer1 & 1];
+    }
+
+    this->timer1++;
+}
+
+void func_809670AC(Demo6K* this, GlobalContext* globalCtx) {
+    this->timer2++;
+
+    if (this->timer1 < 10) {
+        this->timer1++;
+    } else if (this->actor.scale.x > 0.0f) {
+        this->actor.scale.x -= 1.0f / 120.0f;
+        Actor_SetScale(&this->actor, this->actor.scale.x);
+    } else {
+        Actor_Kill(&this->actor);
+    }
+}
+
+void func_8096712C(Demo6K* this, GlobalContext* globalCtx) {
+    static u16 D_8096932C[] = { 275, 275, 275, 275, 275, 275 };
+    u32 frames = globalCtx->state.frames;
+
+    if (this->actor.scale.x < 0.1f) {
+        this->actor.scale.x += 0.0017f;
+    } else if (frames & 1) {
+        this->actor.scale.x = 0.1f * 1.04f;
+    } else {
+        this->actor.scale.x = 0.1f;
+    }
+
+    if ((globalCtx->csCtx.state != CS_STATE_IDLE) && (globalCtx->csCtx.npcActions[6] != NULL) &&
+        (globalCtx->csCtx.npcActions[6]->action == 2)) {
+        Demo6K_SetupAction(this, func_809670AC);
+        this->timer1 = 0;
+        this->actor.scale.x = 0.1f;
+    }
+
+    Actor_SetScale(&this->actor, this->actor.scale.x);
+
+    this->timer2++;
+
+    if ((globalCtx->sceneNum == SCENE_GANONTIKA) && (globalCtx->csCtx.frames < D_8096932C[this->actor.params - 3])) {
+        func_8002F974(&this->actor, NA_SE_EV_LIGHT_GATHER - SFX_FLAG);
+    }
+}
+
+void func_80967244(Demo6K* this, GlobalContext* globalCtx) {
+    static Vec3f velocity = { 0.0f, 0.0f, 0.0f };
+    static Vec3f accel = { 0.0f, 0.0f, 0.0f };
+    static Color_RGBA8 primColor = { 255, 255, 255, 0 };
+    static Color_RGBA8 envColor = { 255, 150, 0, 0 };
+    Vec3f pos;
+    s16 rand1;
+    s16 rand2;
+    s32 scale;
+
+    pos.x = this->actor.world.pos.x;
+    pos.y = this->actor.world.pos.y;
+    pos.z = this->actor.world.pos.z;
+
+    rand1 = Rand_ZeroFloat(0xFFFF);
+    rand2 = Rand_ZeroFloat(0xFFFF);
+
+    velocity.x = Math_SinS(rand2) * Math_CosS(rand1) * 20.0f;
+    velocity.z = Math_CosS(rand2) * Math_CosS(rand1) * 20.0f;
+    velocity.y = Math_SinS(rand1) * 20.0f;
+
+    accel.y = 0.0f;
+
+    envColor.r = sEnvColors[this->unk_293].r;
+    envColor.g = sEnvColors[this->unk_293].g;
+    envColor.b = sEnvColors[this->unk_293].b;
+
+    if (globalCtx->sceneNum == SCENE_TOKINOMA) {
+        scale = 6000;
+    } else if (globalCtx->csCtx.frames < 419) {
+        scale = 6000;
+    } else {
+        scale = 18000;
+    }
+
+    EffectSsKiraKira_SpawnFocused(globalCtx, &pos, &velocity, &accel, &primColor, &envColor, scale, 20);
+}
+
+void func_80967410(Demo6K* this, GlobalContext* globalCtx) {
+    s32 params = this->actor.params - 14;
+
+    this->timer2++;
+
+    Actor_SetScale(&this->actor, 0.2f);
+
+    if ((globalCtx->csCtx.state != CS_STATE_IDLE) && (globalCtx->csCtx.npcActions[params] != NULL)) {
+        func_809691BC(this, globalCtx, params);
+
+        if (globalCtx->csCtx.npcActions[params]->action == 3) {
+            this->flags &= 0xFFFE;
```

Comment:
```suggestion
            this->flags &= ~1;
```

Reply:
gonna go ahead and commit this directly since its such a small change and the only one